dcos/dcos-ui
# DCOS-50452: [1.13] UI Form deletes networkNames property from endpoints for Pods

Add networkNames support for pods in JSON.

Closes https://jira.mesosphere.com/browse/DCOS-50452

## Testing
1. Go to services tab
2. Create a pod
3. Go to Networking tab
4. Change Network Type to "Virtual Network: dcos"
5. Add an endpoint
6. Add a networkNames array in the JSON that has one element "dcos"
7. Verify that the property doesn't disappear when clicking in the form, when going to review and run and back and when running the service (when running it, it is updated after a short delay, so it might not show up immediately)

Example JSON:
```
{
  "id": "/123",
  "containers": [
    {
      "name": "container-1",
      "resources": {
        "cpus": 0.1,
        "mem": 128
      },
      "endpoints": [
        {
          "name": "123",
          "networkNames": ["dcos"],
          "containerPort": 123,
          "hostPort": 0,
          "protocol": [
            "tcp"
          ]
        }
      ]
    }
  ],
  "scaling": {
    "instances": 1,
    "kind": "fixed"
  },
  "networks": [
    {
      "name": "dcos",
      "mode": "container"
    }
  ],
  "volumes": [],
  "fetch": [],
  "scheduling": {
    "placement": {
      "constraints": []
    }
  }
}
```

## Trade-offs
None.

## Dependencies
None.

## Screenshots

### Before
![Peek 2019-03-20 13-44](https://user-images.githubusercontent.com/40791275/54682503-c7354980-4b17-11e9-99b4-d30f494e7fa1.gif)

### After
![Peek 2019-03-20 13-45](https://user-images.githubusercontent.com/40791275/54682523-d1574800-4b17-11e9-9863-9797e260dd96.gif)
